cli/failover/setup.go

```diff
@@ -0,0 +1,108 @@
+package failover
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/apex/log"
+	"github.com/fatih/structs"
+	"github.com/tarantool/cartridge-cli/cli/common"
+	"github.com/tarantool/cartridge-cli/cli/connector"
+	"github.com/tarantool/cartridge-cli/cli/context"
+	"github.com/tarantool/cartridge-cli/cli/replicasets"
+	"gopkg.in/yaml.v2"
+)
+
+type FailoverOpts struct {
+	Mode             string          `yaml:"mode" structs:"mode"`
+	StateProvider    *string         `yaml:"state_provider,omitempty" structs:"state_provider"`
+	StateboardParams *ProviderParams `yaml:"stateboard_params,omitempty" structs:"tarantool_params"`
+	Etcd2Params      *ProviderParams `yaml:"etcd2_params,omitempty" structs:"etcd2_params"`
+
+	FailoverTimeout *int  `yaml:"failover_timeout,omitempty" json:"failover_timeout" structs:"failover_timeout"`
+	FencingEnabled  *bool `yaml:"fencing_enabled,omitempty" json:"fencing_enabled" structs:"fencing_enabled"`
+	FencingTimeout  *int  `yaml:"fencing_timeout,omitempty" json:"fencing_timeout" structs:"fencing_timeout"`
+	FencingPause    *int  `yaml:"fencing_pause,omitempty" json:"fencing_pause" structs:"fencing_pause"`
+}
+
+type ProviderParams map[string]interface{}
+
+func (failoverOpts *FailoverOpts) Manage(ctx *context.Ctx) error {
```

Comment:
Function `Manage` and `FailoverOpts` aren't specific for `setup` command, this file should be called `manage.go`.